allennlp/data/iterators/same_lang_iterator.py

```diff
@@ -0,0 +1,46 @@
+from collections import deque, defaultdict
+from typing import Iterable, Deque
+import logging
+import random
+
+from allennlp.common.util import lazy_groups_of
+from allennlp.data.instance import Instance
+from allennlp.data.iterators.data_iterator import DataIterator
+from allennlp.data.dataset import Batch
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+def split_by_lang(instance_list):
+    insts_by_lang = defaultdict(lambda: [])
+    for inst in instance_list:
+        inst_lang = inst.fields['metadata'].metadata['lang']
+        insts_by_lang[inst_lang].append(inst)
+
+    return iter(insts_by_lang.values())
+
+@DataIterator.register("same_lang")
+class SameLangIterator(DataIterator):
```

Comment:
Also, I tried hard not to care about `lang` versus `language` in your data and model code, because it should be pretty obvious based on context what the variable names mean there.  But if I'm just looking at the data iterators, I don't have the context of "multi-lingual" at all, so calling this `SameLanguageIterator` would be better here (this applies to the filename also).